pkg/base1/test-dbus-common.js

```diff
@@ -932,12 +932,12 @@ export function common_dbus_tests(channel_options, bus_name) { // eslint-disable
                             "com.redhat.Cockpit.DBusTests.Hidden": { Name: name }
                         }, "got data before signal");
                         dbus.removeEventListener("notify", onnotify);
-                        done();
                     });
                     dbus.call("/otree/frobber", "com.redhat.Cockpit.DBusTests.Frobber",
                               "EmitHidden", [name])
                             .always(function() {
                                 assert.equal(this.state(), "resolved", "method called");
+                                done();
```

Comment:
This looks wrong. See src/ws/mock-service.c, the EmitHidden() call is what *causes* the DBusTests.Hidden signal to be sent, so it's correct to wait for it and only succeed the test in the signal handler above.

If this fails, it means that the signal wasn't sent, or not received.

Reply:
from IRC: The EmitHidden function is implemented in a way that emits the signal before returning. The subscription would get a callback before the promise on the call came back

Reply:
Looking at `emit_hidden()` in `mock-service.c` it's pretty clear that

```
  g_dbus_connection_emit_signal (connection, NULL, path,
                                 "com.redhat.Cockpit.DBusTests.Hidden", "Yooohooo",
                                 g_variant_new ("()"), &error);
```

gets called before

```
  test_frobber_complete_emit_hidden (frobber, invocation);
```

which means we'd indeed expect to see the signal before the method returns.